cEP-0024.md

```diff
@@ -0,0 +1,255 @@
+adds plugins to auto-review, auto-reject, auto-label PRs, assign, unassign
+issues, squash commits.
+
+## Changes in coala workflow
+
+### Use increasing difficulty level order to assign issues
+
+The current assign process in coala is to check team membership of a user,
+difficulty level of the issue and assign the issue to user if eligible. To
+implement the current assign process in GitMate, IGitt would need to support
+teams.
+
+Currently, IGitt doesn't support teams and there are no teams in GitLab.
+Considering [GitLab groups](https://docs.gitlab.com/ee/user/group/) and
+[GitHub teams](https://help.github.com/articles/about-teams/) as teams in
+IGitt is a possible solution for IGitt implementation. But, it requires extra
+tools to sync GitHub team and GitLab group memberships, which is a manual
+process and isn't automated yet.
+
+Also, team membership information is not publicly accessible in GitHub, so any
+system which needs to implement the current assign system would require
+organization authentication tokens.
+
+The best possible solution would be using an increasing difficulty level order,
+so that a user can be assigned to an issue only if the user has solved issues
+of lower difficulty level before requesting to get assigned to a higher
+difficulty level issue.
+
+#### Drawbacks
+
+If a user completed the lower difficulty level issues in GitHub and want to
+work on a higher difficulty level issue in GitLab, he/she has to repeat the
+newcomer process in GitLab. To resolve this issue, GitMate needs to find a way
+to link GitHub org and GitLab org, this would be easy when they have the same
+name, as in the case of `coala`. But, resolving this issue is out of scope for
+this project.
+
+If a user is worthy enough to work on the higher level difficulty issues
+directly, then that user needs to be manually assigned by the maintainers until
+that user completes at least one low difficulty issue. Even with current assign
+system, manual intervention by maintainers is needed to add a user to
+developers team.
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened or updated, if any of PR's commit message referenced
+any issues with keywords `Fixes` or `Closes`, there is no
+CI failure in PR, then auto assign the author of the PR to the mentioned
+unassigned issues. If the author of the PR isn't eligible to get assigned to
+the mentioned issues, missing eligibility conditions will be specified as a
+comment, which isn't configurable.
+
+If someone else is working on the mentioned issue, GitMate will
+comment (configurable) on the PR.
+
+If the author of a PR is not a member of the organization, then he/she can't be
+assigned to any of the issues, a label (configurable) will be added on the PR
+so that maintainers can identify these type of PRs and GitMate will comment
+on the PR informing the author of the PR to get invited to the organization
```

Comment:
after the newcomer has accepted the invite, when/how will the label on their PR be removed?

Will any comment / review on the PR cause the label to be removed?

Or do they need to re-push to get their label updated?

Or does a maintainer have to remove it manually?

Reply:
Maintainer have to remove it manually

Reply:
Could it be removed automatically on the next push ?